validation/src/header.rs

```diff
@@ -203,35 +200,46 @@ fn find_next_difficulty_in_chain(
     store: &impl HeaderStore,
     prev_header: &BlockHeader,
     prev_height: BlockHeight,
-) -> u32 {
+) -> (u32, u32) {
```

Comment:
IIUC, here we're exposing some internal information, the number of headers inspected, to verify it in our tests. I wonder if this is necessary. Here's my reasoning:

1. If it's necessary, then that means that our coverage of this method is insufficient, since we should be able to come up with examples where the difficulty is wrong if the number of headers is wrong.
2. If it's not necessary, then we have enough coverage for this method, so why have it at all?

Reply:
Removed `headers_inspected` counter.